docs/high/dataset.rst

```diff
@@ -508,10 +508,23 @@ Reference
         type.  Conversion is handled by HDF5 directly, on the fly::
 
             >>> dset = f.create_dataset("bigint", (1000,), dtype='int64')
-            >>> out = dset.astype('int16')[:]
+            >>> out = dset.astype('int16')[:500]
             >>> out.dtype
             dtype('int16')
 
+        This is typically faster than reading the data and then
+        converting it with NumPy:
+
+            >>> out = dset[:500].astype('int16')  # slower
+
+        Calling ``.astype('T')`` (NumPy's native variable-width strings)
+        on a dataset returns a writable dataset with the specified dtype;
+        read more at :ref:`npystrings`.
+        For all other dtypes, this method returns a read-only view.
```

Comment:
Why is a writable view needed for this case? I think writing an array with "T" dtype into a variable length string dataset should work without needing a view.

Reply:
Yes, writing a StringDType array into a vlen Dataset works.
However, if the user fancies _reading_ back from it, they would be surprised to get an object array.
Particularly (see my previous comment) if they just explicitly called `create_dataset(..., dtype="T")`.

Reply:
I missed changing the docs, but since my changes that you merged into this PR, `.astype('T')` is now more like other `.as*` methods, giving you a read-only view, while you can write without needing astype.